extensions/word-search/src/components/SearchResults.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+
+import type { LaunchProps } from "@raycast/api";
+import { Action, List } from "@raycast/api";
+
+import type { SearchType } from "@/types";
+import { Vocabulary } from "@/types";
+
+import useOptionalSelection from "@/hooks/use-optional-selection";
+import useSearchWords from "@/hooks/use-searchwords";
+
+import Actions from "@/components/Actions";
+import VocubalarySwitch from "@/components/VocubalarySwitch";
+
+export interface extraOptions {
+  useVocabulary?: boolean;
+  helperTitle?: string;
+  helperDescription?: string;
+}
+
+export default function SearchResults(
+  type: SearchType,
+  placeholder: string,
+  launchProps: LaunchProps,
+  { helperTitle, helperDescription, useVocabulary }: extraOptions = {},
+) {
+  const [search, setSearch] = useState<string>("");
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+  const [vocubalary, setVocubalary] = useState<Vocabulary>(Vocabulary.English);
+
+  useOptionalSelection(setSearch, typeof launchProps.fallbackText !== "undefined" && launchProps.fallbackText !== "");
+
+  const { data, isLoading } = useSearchWords(search, type, useVocabulary ? vocubalary : undefined);
+
+  return (
+    <List
+      searchBarPlaceholder={placeholder}
+      isLoading={isLoading}
+      throttle={true}
+      onSearchTextChange={setSearch}
+      isShowingDetail={search !== "" && showDetails}
+      searchBarAccessory={useVocabulary ? <VocubalarySwitch onChange={setVocubalary} /> : null}
+      searchText={search}
+    >
+      {!data || data.length === 0 ? (
+        <List.EmptyView
+          icon={"command-icon.png"}
+          title={helperTitle ?? placeholder}
+          description={search !== "" ? (isLoading ? "Searching..." : "No Results Found") : helperDescription}
+        />
+      ) : (
+        data.map((word) => (
+          <List.Item
+            icon={"command-icon.png"}
+            key={word.word}
+            title={word.word}
+            subtitle={word.defs !== undefined ? word.defs[0] : ""}
+            actions={
+              <Actions word={word}>
+                <Action
+                  title="Toggle Details"
+                  onAction={() => setShowDetails((d) => !d)}
+                  shortcut={{ key: "d", modifiers: ["cmd"] }}
+                />
+              </Actions>
+            }
+            detail={
+              <List.Item.Detail
+                markdown={
+                  word.defs !== undefined
+                    ? "```\n" + word.defs.map((d) => d.replaceAll("n\t", "")).join("\n\n") + "```"
```

Comment:
I think having each definition in its own ``` might look better!
![Screenshot 2024-06-02 at 22 18 00@2x](https://github.com/raycast/extensions/assets/15897093/cc8a524e-ee69-47e7-95ff-6134ee9cf1b9)

What do you think?

```suggestion
                    ? word.defs.map((d) => "```\n" + d.replaceAll("n\t", "") + "\n```").join("\n\n")
```
With this, it looks cleaner. Also, I think there might be a better way to present the tags, because right now they take up half the space. Maybe just an added text at the bottom of the definitions? Or maybe there's a different much better way to present it? What do you think?